### PR TITLE
win32_deps_build: avoid pip

### PR DESCRIPTION
**Pull request by petrutlucian94**
We're currently installing cython with pip when using Ubuntu to cross compile Ceph for Windows. This can fail with recent Python versions if attempting to use the global env:

```
  error: externally-managed-environment

  × This environment is externally managed
  ╰─> To install Python packages system-wide, try apt install
      python3-xyz, where xyz is the package you are trying to
      install.
```

Cython isn't really needed by the Windows build so we can go ahead and drop it. We were hoping to use the Python bindings on Windows, however Python extensions can't be cross compiled.

While at it, we'll also ensure that git is installed.

Fixes: https://tracker.ceph.com/issues/59354

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: win32_deps_build.sh [hunk 1]
```diff
@@ -88,8 +88,8 @@ case "$OS" in
         sudo env DEBIAN_FRONTEND=noninteractive apt-get -y install \
             mingw-w64 cmake pkg-config \
             python3-dev python3-pip python3-yaml \
-                autoconf libtool ninja-build wget zip
-        sudo python3 -m pip install cython
```
Comment (idryomov) [inline win32_deps_build.sh:92]: Is `python3-pip` above still needed then?

**Comment by idryomov** [comment]
- Can you link where this popped up?
- Would installing Cython from the distro repo work for Focal or is this specific to Jammy?

**Comment by petrutlucian94** [comment]
> * Can you link where this popped up?
> * Would installing Cython from the distro repo work for Focal or is this specific to Jammy?

This error showed up while cross compiling using a Ubuntu 23.04 container. The cython3 package is also available for Focal: https://launchpad.net/ubuntu/focal/+package/cython3

**Comment by idryomov** [comment]
> The cython3 package is also available for Focal: https://launchpad.net/ubuntu/focal/+package/cython3

It's also available for Bionic, but an older version (0.26 vs 0.29): https://packages.ubuntu.com/bionic/cython3.  Was that why you went with pip initially?

**Comment by petrutlucian94** [comment]
On a second thought, we can actually drop the Cython dependency. We were trying to port the python bindings to Windows, however python extensions can't be cross compiled due to ABI incompatibilities.

About recent CI failures, I noticed that some tests occasionally time out. I think we're hitting winpthreads mutex issues again: 

* https://paste.opendev.org/raw/bRJeqkMVLHOKthJUeHUc/
* https://github.com/ceph/ceph/pull/46989

I've tried switching to mingw/llvm and it seems to fix the issue. libc++ doesn't use winpthreads, unlike libstdc++ with mingw/gcc.

**Comment by petrutlucian94** [comment]
Reverting this PR fixes the connection issue: https://github.com/ceph/ceph/pull/47457. I'll need to figure out the exact cause.

**Comment by idryomov** [comment]
> Reverting this PR fixes the connection issue: #47457.

... and it didn't pass the Windows check, of course.  I'll push for making it required once this latest set of issues/regressions is resolved.

**Comment by petrutlucian94** [comment]
> > Reverting this PR fixes the connection issue: #47457.
> 
> ... and it didn't pass the Windows check, of course. I'll push for making it required once this latest set of issues/regressions is resolved.

That would be extremely helpful, I'll try to address the issues today.

**Comment by petrutlucian94** [comment]
I've submitted a patch: https://github.com/ceph/ceph/pull/51043.

**Comment by idryomov** [comment]
jenkins test windows

**Comment by idryomov** [comment]
The build failure (https://github.com/ceph/ceph/pull/50916#issuecomment-1503692904) still persists.

**Comment by petrutlucian94** [comment]
> The build failure [#50916 (comment)](https://github.com/ceph/ceph/pull/50916#issuecomment-1503692904) still persists.

Fixed it, thanks. g++ was getting installed as a pip dependency and now has to be specified explicitly. It's used when building the boost b2 tool.

**Review by idryomov — APPROVED** [review]

**Pull request merged**